packages/components/src/styling/src/BreakpointProvider.tsx

```diff
@@ -37,6 +36,8 @@ export function BreakpointProvider({
     children,
     unsupportedMatchMediaBreakpoint = "lg"
 }: BreakpointProvider) {
+    // Ensure that matchMedia function exists. In a jest environement or in SSR, this function is not available.
+    const supportsMatchMedia = typeof window !== "undefined" && typeof window.matchMedia === "function";
```

Comment:
Why not keep this keep this check is `useMediaQuery` ? Also, it should use `isNil` and `isFunction`

Reply:
We can not use isNil, since we don't check the value of window (if we try to access the value of window, it breaks) so we check the type of it. isNil would not work

Reply:
Got it, still why not keep it in `useMediaQuery` ?